### PR TITLE
Bugfix/5271 wrong availability displayed on nov 6

### LINKED ISSUE #5271: [CAL-292] Displays the wrong availability
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Accessed a user's event link who's situated in PST. I am located in EST. It displayed the wrong availability

### Steps to Reproduce

1. Clicked on the event link of a user who is in Pacific Time
2. It was displayed the user is available starting at 11am EST (which is 8am PT), but the user availability start at 9am PT / 12 EST/

### Technical details

Self Hosted, main branch v 2.1.1

> From [Linear-GitHub Sync](https://synclinear.com)

### PR DESCRIPTION
**Pull request by emrysal**
Fixes #5271 

* Fixes DST day bug.
* Fixes time formatting on the booking page.

### CODE REVIEW

File: apps/web/components/booking/AvailableTimes.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const AvailableTimes: FC<AvailableTimesProps> = ({
               pathname: router.pathname.endsWith("/embed") ? "../book" : "book",
               query: {
                 ...router.query,
-                date: dayjs(slot.time).format(),
+                date: dayjs.utc(slot.time).tz(timeZone()).format(),
```
Comment (emrysal) [inline apps/web/components/booking/AvailableTimes.tsx:86]: slot.time comes in as UTC, and dayjs() automatically converted this to browser time. We don't want that, we want to submit to user preferences.

File: apps/web/lib/parseDate.ts [hunk 2]
```diff
@@ -20,15 +20,6 @@ export const parseDate = (date: string | null | Dayjs, i18n: I18n) => {
   return processDate(date, i18n);
 };
 
-// tzid is currently broken in rrule library.
-// @see https://github.com/jakubroztocil/rrule/issues/523
-const dateWithZone = (d: Date, timeZone?: string) => {
```
Comment (emrysal) [inline apps/web/lib/parseDate.ts:25]: using utcOffset instead

File: apps/web/lib/parseDate.ts [hunk 3]
```diff
@@ -48,13 +39,18 @@ export const parseRecurringDates = (
...
+    // applying the DST offset.
+    return dayjs.utc(t).add(startUtcOffset - dayjs(t).utcOffset(), "minute");
   });
-  // UTC times with tzOffset applied to account for DST
-  const times = rule.all().map((t) => dateWithZone(t, timeZone));
   const dateStrings = times.map((t) => {
-    // undo DST diffs for localized display.
-    return processDate(dayjs.utc(t).tz(timeZone), i18n);
+    // finally; show in local timeZone again
+    return processDate(t.tz(timeZone), i18n);
```
Comment (emrysal) [inline apps/web/lib/parseDate.ts:53]: t is already in UTC mode

**Review by sean-brydon — APPROVED** [review]
LGTM - tested on prod to replicate bugs and they were solved locally :) 

Thanks for stealing me @emrysal 🎉

**Pull request merged**